engine/execution/state/delta/view.go

```diff
@@ -253,8 +250,15 @@ func (v *View) ReadsCount() uint64 {
 }
 
 // SpockSecret returns the secret value for SPoCK
+//
+// This function modifies the internal state of the SPoCK secret hasher.
+// Once called, it doesn't allow writing more data into the SPoCK secret.
 func (v *View) SpockSecret() []byte {
-	return v.spockSecretHasher.SumHash()
+	// check if spocksSecret has been already computed
+	if v.spocksSecret == nil {
+		v.spocksSecret = v.spockSecretHasher.SumHash()
+	}
+	return v.spocksSecret
```

Comment:
Are we sure that this does not break any concurrency?

Reply:
Not sure if we the function needs to be thread safe, but I added a lock in case it needs to be.